low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -0,0 +1,168 @@
+
+    def __str__(self):
+        if self._rc == 0: return self._desc
+        return "error(%d): %s" %(self._rc, self._desc)
+
+
+class SSPLPostInstall:
+    def __init__(self, args: list):
+        """Ssplpostinstall init."""
+        self.args = args
+        self.name = "sspl_post_install"
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+        self.RSYSLOG_CONF="/etc/rsyslog.d/0-iemfwd.conf"
+        self.RSYSLOG_SSPL_CONF="/etc/rsyslog.d/1-ssplfwd.conf"
+        self.PACEMAKER_INSTALLATION_PATH="/lib/ocf/resource.d/seagate/"
+
+    def process(self):
+        if not self.args or len({'-p', '-e', '-c'} & set(self.args)) == 0:
+            raise PostInstallError(errno.EINVAL, f"Invalid Argument for {self.name}")
+
+        PRODUCT = ""
+        ENVIRONMENT = ""
+        RMQ_CLUSTER = "true"
+        i = 0
+        while i < len(self.args):
+            try:
+                if self.args[i] == '-p':
+                    PRODUCT = self.args[i+1]
+                elif self.args[i] == '-e':
+                    ENVIRONMENT = self.args[i+1]
+                elif self.args[i] == '-c':
+                    RMQ_CLUSTER = self.args[i+1]
+            except (IndexError, ValueError):
+                raise PostInstallError(errno.EINVAL, f"Invalid Argument for {self.name}")
+            i+=1
+
+        # sspl_setup_consul script install consul in dev env and checks if consul process is running
+        # on prod. For node replacement scenario consul will not be running on the new node. But,
+        # there will be two instance of consul running on healthy node. When new node is configured
+        # consul will be brought back on it. We are using VIP to connect to consul. So, if consul
+        # is not running on new node, we dont need to error out. So, need to skip this step for
+        # node replacement case
+        # Onward LDR_R2, consul will be abstracted out and it won't exit as hard dependeny of SSPL
+        if PRODUCT_NAME == "LDR_R1":
+            # setup consul if not running already
+            if not os.path.exists(REPLACEMENT_NODE_ENV_VAR_FILE):
+                sspl_setup_consul = f"{self._script_dir}/sspl_setup_consul -e {ENVIRONMENT}"
+                output, error, returncode = SimpleProcess(sspl_setup_consul).run()
+                if returncode != 0:
+                    raise PostInstallError(returncode, error, sspl_setup_consul)
+
+        # Install packages which are not available in YUM repo, from PIP
+        pip_cmd = f"python3 -m pip install -r {SSPL_BASE_DIR}/low-level/requirements.txt"
+        output, error, returncode = SimpleProcess(pip_cmd).run()
+        if returncode != 0:
+            raise PostInstallError(returncode, error, pip_cmd)
+        # Splitting current function into 2 functions to reduce the complexity of the code.
+        self.make_file_operations(PRODUCT, ENVIRONMENT, RMQ_CLUSTER)
+
+    def make_file_operations(self, PRODUCT, ENVIRONMENT, RMQ_CLUSTER):
```

Comment:
Let's rename this something like `install_files`

Reply:
Done

Reply:
DOne